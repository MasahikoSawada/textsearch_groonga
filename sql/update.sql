CREATE TABLE item (name text, counter integer) WITH (fillfactor = 50);
INSERT INTO item VALUES ('foo', 0);
INSERT INTO item SELECT 'bar', 0 FROM generate_series(1, 999);
CREATE INDEX item_idx ON item USING groonga (name);
ANALYZE item;

SET enable_seqscan = off;
SELECT * FROM item WHERE name %% 'foo';
UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
VACUUM item;
SELECT * FROM item WHERE name %% 'foo';
UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
VACUUM item;
SELECT * FROM item WHERE name %% 'foo';
UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
RESET enable_seqscan;
