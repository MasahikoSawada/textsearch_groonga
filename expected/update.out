CREATE TABLE item (name text, counter integer) WITH (fillfactor = 50);
INSERT INTO item VALUES ('foo', 0);
INSERT INTO item SELECT 'bar', 0 FROM generate_series(1, 999);
CREATE INDEX item_idx ON item USING groonga (name);
ANALYZE item;
SET enable_seqscan = off;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       0
(1 row)

UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       1
(1 row)

UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       2
(1 row)

UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       3
(1 row)

UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       4
(1 row)

VACUUM item;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       4
(1 row)

UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       5
(1 row)

VACUUM item;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       5
(1 row)

UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       6
(1 row)

UPDATE item SET counter = counter + 1;
SELECT * FROM item WHERE name %% 'foo';
 name | counter 
------+---------
 foo  |       7
(1 row)

RESET enable_seqscan;
